### PR TITLE
Code and project cleanup

### PR DESCRIPTION
**Pull request by manigandham**

### CODE REVIEW

File: src/Hashids.net/Hashids.net.csproj [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
+	<ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
+		<PackageReference Include="System.Buffers" Version="4.5.1" />
+	</ItemGroup>
 
-    <ItemGroup>
-      <PackageReference Include="System.Memory" Version="4.5.4" />
-    </ItemGroup>
+	<ItemGroup>
+		<PackageReference Include="System.Memory" Version="4.5.4" />
+	</ItemGroup>
```
Comment (KeterSCP) [inline src/Hashids.net/Hashids.net.csproj:32]: System.Memory package reference should also be conditional only for `netstandard2.0`

File: test/Hashids.net.test/Hashids.net.test.csproj [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-        <PackageReference Include="xunit" Version="2.4.1" />
-        <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3">
-            <PrivateAssets>all</PrivateAssets>
-            <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
-        </PackageReference>
-    </ItemGroup>
+	<ItemGroup>
+		<PackageReference Include="FluentAssertions" Version="5.10.3" />
+		<PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.9.4" />
+		<PackageReference Include="Moq" Version="4.16.1" />
```
Comment (KeterSCP) [inline test/Hashids.net.test/Hashids.net.test.csproj:13]: Can we update these packages in this PR?

Comment (manigandham) [reply test/Hashids.net.test/Hashids.net.test.csproj:13]: Will do

**Comment by KeterSCP** [comment]
Should this also be refactored? https://github.com/ullmark/hashids.net/blob/311839f9f8af119d84c11462c02e43e45b17ba98/src/Hashids.net/Hashids.cs#L72

**Comment by manigandham** [comment]
> Should this also be refactored?
> 
> https://github.com/ullmark/hashids.net/blob/311839f9f8af119d84c11462c02e43e45b17ba98/src/Hashids.net/Hashids.cs#L72

Yes, that's done.

File: src/Hashids.net/Hashids.cs [hunk 3]
```diff
@@ -54,50 +54,35 @@ public Hashids() : this(salt: string.Empty, minHashLength: 0, alphabet: DEFAULT_
...
 
-        /// <remarks>This method uses <c>out</c> params instead of returning a ValueTuple so it works with .NET 4.6.1.</remarks>
-        private static void InitCharArrays(string alphabet, string seps, ReadOnlySpan<char> salt, out char[] alphabetChars, out char[] sepChars, out char[] guardChars)
+        private static (char[] alphabetChars, char[] sepChars, char[] guardChars) InitCharArrays(string alphabet, string seps, ReadOnlySpan<char> salt)
         {
-            alphabetChars = alphabet.ToCharArray().Distinct().ToArray();
-            sepChars      = seps.ToCharArray();
+            var alphabetChars = alphabet.ToCharArray().Distinct().ToArray();
+            var sepChars = seps.ToCharArray();
+            var guardChars = new char[0];
```
Comment (KeterSCP) [inline src/Hashids.net/Hashids.cs:71]: There is no need for empty allocation here, it should just be

```c#
char[] guardChars;
```

**Comment by manigandham** [comment]
Some slight performance improvements. Considering this to be more than enough optimization at this point.

```
|               Method |        Job |            Runtime |       Mean |    Error |   StdDev |  Gen 0 | Code Size | Allocated |
|--------------------- |----------- |------------------- |-----------:|---------:|---------:|-------:|----------:|----------:|
|        RoundtripInts | Job-XEDAQS |           .NET 5.0 | 3,642.3 ns | 71.33 ns | 70.06 ns | 0.0610 |      50 B |     512 B |
|       RoundtripLongs | Job-XEDAQS |           .NET 5.0 | 4,161.7 ns | 72.26 ns | 67.59 ns | 0.0610 |   3,310 B |     512 B |
|         RoundtripHex | Job-XEDAQS |           .NET 5.0 | 3,554.4 ns | 67.29 ns | 66.08 ns | 0.1602 |      49 B |   1,344 B |
|         SingleNumber | Job-XEDAQS |           .NET 5.0 |   795.3 ns | 15.07 ns | 16.75 ns | 0.0076 |     153 B |      64 B |
| SingleNumberAsParams | Job-XEDAQS |           .NET 5.0 |   842.3 ns | 12.04 ns | 11.26 ns | 0.0191 |   2,059 B |     160 B |
|                      |            |                    |            |          |          |        |           |           |
|        RoundtripInts | Job-FYFVBK |           .NET 6.0 | 3,596.6 ns | 61.46 ns | 57.49 ns | 0.0610 |      50 B |     512 B |
|       RoundtripLongs | Job-FYFVBK |           .NET 6.0 | 4,049.1 ns | 80.22 ns | 95.50 ns | 0.0610 |   3,256 B |     512 B |
|         RoundtripHex | Job-FYFVBK |           .NET 6.0 | 3,338.9 ns | 20.16 ns | 18.86 ns | 0.1602 |      49 B |   1,344 B |
|         SingleNumber | Job-FYFVBK |           .NET 6.0 |   772.8 ns | 12.47 ns | 11.66 ns | 0.0076 |     145 B |      64 B |
| SingleNumberAsParams | Job-FYFVBK |           .NET 6.0 |   852.2 ns | 16.66 ns | 16.36 ns | 0.0191 |   2,042 B |     160 B |
|                      |            |                    |            |          |          |        |           |           |
|        RoundtripInts | Job-TPPQKV | .NET Framework 4.8 | 4,022.2 ns | 33.66 ns | 31.48 ns | 0.1221 |      51 B |     794 B |
|       RoundtripLongs | Job-TPPQKV | .NET Framework 4.8 | 4,625.8 ns | 39.48 ns | 36.93 ns | 0.1373 |   4,993 B |     907 B |
|         RoundtripHex | Job-TPPQKV | .NET Framework 4.8 | 5,241.2 ns | 55.17 ns | 51.60 ns | 0.2747 |      52 B |   1,781 B |
|         SingleNumber | Job-TPPQKV | .NET Framework 4.8 |   982.6 ns | 19.00 ns | 21.12 ns | 0.0362 |   3,240 B |     241 B |
| SingleNumberAsParams | Job-TPPQKV | .NET Framework 4.8 |   992.3 ns | 12.64 ns | 11.82 ns | 0.0362 |   3,240 B |     241 B |
```

**Pull request merged**